ksqldb-execution/src/main/java/io/confluent/ksql/execution/plan/ExecutionStep.java

```diff
@@ -31,7 +31,8 @@
     @Type(value = StreamAggregate.class, name = "streamAggregateV1"),
     @Type(value = StreamFilter.class, name = "streamFilterV1"),
     @Type(value = StreamFlatMap.class, name = "streamFlatMapV1"),
-    @Type(value = StreamGroupBy.class, name = "streamGroupByV1"),
+    @Type(value = StreamGroupByV1.class, name = "streamGroupByV1"),
+    @Type(value = StreamGroupBy.class, name = "streamGroupByV2"),
```

Comment:
to avoid confusion that I have with `StreamSelectKey` every time i look at the class, what do you think about (in a follow-up PR) changing the class name of `StreamGroupBy` to `StreamGroupByV2` and `StreamSelectKey` to `StreamSelectKeyV2`

Reply:
I actually rather like the convention that the non-versioned class is always the latest. It makes it easier to find the latest code rather than needing to first check/reason about what the latest version is. 

Would you want to rename just the execution step classes or also the affiliated builders and execution plan traversal visitor methods? If the confusion is just around the execution step classes themselves, I could be on-board for that rename. I'm not sure renaming the others saves more confusion than is worth the toil it creates but I don't feel too strongly.

Reply:
fair enough, I don't feel _too_ strongly and I guess I can see it either way